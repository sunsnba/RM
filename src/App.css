body { margin: 0 }


.HeadStyle {
  height: 105px;
  border: solid;
  display: flex;
  flex-wrap: nowrap;
  color: black;
  justify-content: space-around;
  align-items: center;
}


.A {
  margin-top: 2%;
  margin-left: 70px;
  width: 600px;
  height: 600px;
  border: dotted; 
  background: #FFFFFF;
  position: absolute;
  text-align: center;
  vertical-align: middle;

}

.B {
  margin-top: 2%;
  margin-left: 760px;
  width: 600px;
  height: 600px;
  border: dotted; 
  background: #EEEEEE;
  position: absolute;
}

.plus-button {
  border: none;
  background: none;
  font-size: 15px;
  margin-left: 1125px;
}

.minus-button {
  border: none;
  background: none;
  font-size: 15px;

}

.C {
  margin-top: 2%;
  margin-left: 760px;
  width: 600px;
  height: 600px;
  border: dotted; 
  background:  #FFFFFF;
  position: absolute;
  text-align: center;
  font-size: 16px;
}

.D {
  margin-top: 2%;
  margin-left: 70px;
  width: 600px;
  height: 600px;
  border: dotted; 
  background: #EEEEEE;
  position: absolute;
}