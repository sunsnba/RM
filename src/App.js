import React, { Component } from 'react';
import './App.css';
import Header from './Header';
import PanelA from './PanelA';
import PanelB from './PanelB';
import data from './data.json';

class App extends Component {
  constructor(props) {
    super(props)

  this.state = {
    hData: data.headerData,
    contA: data.contentA,
    contB: data.contentB,
    isPositive: false,
  }
  this.headerClick = this.headerClick.bind(this);    
  }

  headerClick = () => {
    this.setState(prevState => ({
      isPositive: !prevState.isPositive,
    }));
  }

  render() {
    return (
      <div className="App">
        <header className="App-header">
         <Header hData={this.state.hData} headerClick={this.headerClick} isPositive={this.state.isPositive}/>
        </header>
        <PanelA />
        <PanelB />
      </div>
    );
  }
}

export default App;
