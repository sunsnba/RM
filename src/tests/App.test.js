import React from 'react';
import App from '../App'
import renderer from 'react-test-renderer';



it('renders a snapshot', () => {
  const tree = renderer.create(<App/>).toJSON();
  expect(tree).toMatchSnapshot();
});

// note it was necessary to manually install babel-core@7.0.0-bridge.0 for jest tests to properly run