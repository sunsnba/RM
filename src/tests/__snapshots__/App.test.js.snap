// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot 1`] = `
<div>
  <div
    className="boards-container"
  >
    <div
      className="board"
    >
      <div>
         
        <div
          className="HeadStyle"
        >
          <div
            className="HeadContent"
          >
            <ul>
              30%
            </ul>
            <ul>
              $2000000
            </ul>
            <ul>
              85%
            </ul>
            <input
              className="minus-button"
              onClick={[Function]}
              type="button"
              value="-"
            />
          </div>
        </div>
         
      </div>
      <div
        className="PanelContainer"
      >
        <div>
           
          <button
            className="A"
          >
             
            This should be displayed in Panel A. This is visible by default
             
          </button>
           
        </div>
        <div>
           
          <button
            className="B"
            onClick={[Function]}
          >
             
          </button>
           
        </div>
      </div>
    </div>
  </div>
  <div
    className="boards-container"
  >
    <div
      className="board"
    >
      <div>
         
        <div
          className="HeadStyle"
        >
          <div
            className="HeadContent"
          >
            <ul>
              30%
            </ul>
            <ul>
              $2000000
            </ul>
            <ul>
              85%
            </ul>
            <input
              className="minus-button"
              onClick={[Function]}
              type="button"
              value="-"
            />
          </div>
        </div>
         
      </div>
      <div
        className="PanelContainer"
      >
        <div>
           
          <button
            className="A"
          >
             
            This should be displayed in Panel A. This is visible by default
             
          </button>
           
        </div>
        <div>
           
          <button
            className="B"
            onClick={[Function]}
          >
             
          </button>
           
        </div>
      </div>
    </div>
  </div>
  <div
    className="boards-container"
  >
    <div
      className="board"
    >
      <div>
         
        <div
          className="HeadStyle"
        >
          <div
            className="HeadContent"
          >
            <ul>
              30%
            </ul>
            <ul>
              $2000000
            </ul>
            <ul>
              85%
            </ul>
            <input
              className="minus-button"
              onClick={[Function]}
              type="button"
              value="-"
            />
          </div>
        </div>
         
      </div>
      <div
        className="PanelContainer"
      >
        <div>
           
          <button
            className="A"
          >
             
            This should be displayed in Panel A. This is visible by default
             
          </button>
           
        </div>
        <div>
           
          <button
            className="B"
            onClick={[Function]}
          >
             
          </button>
           
        </div>
      </div>
    </div>
  </div>
</div>
`;
